session/session_test.go

```diff
@@ -3145,6 +3145,31 @@ func (s *testSessionSuite2) TestStmtHints(c *C) {
 	c.Assert(tk.Se.GetSessionVars().GetReplicaRead(), Equals, kv.ReplicaReadFollower)
 }
 
+// Test memory track in POINT_GET/BATCH_POINT_GET for issue 21653
+func (s *testSessionSuite2) TestPointGetMemoryTracking(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t1;")
+	tk.MustExec("create table t1(a int primary key);")
+	defer config.RestoreFunc()()
+	config.UpdateGlobal(func(conf *config.Config) {
+		conf.OOMAction = config.OOMActionCancel
+	})
+	tk.MustExec("insert into t1 values (1);")
+	tk.MustExec("select * from t1 where a = 1;")
+	tk.MustExec("set tidb_mem_quota_query=1;")
+	err := tk.QueryToErr("select * from t1 where a = 1;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Matches, "Out Of Memory Quota!.*")
+	err = tk.ExecToErr("update t1 set a = a + 10 where a in (1,2,3);")
```

Comment:
use `select * from t1 where a in  (1,2,3);` is ok...  I think, we should only test BatchPointGet, unless UpdateExec doesn't consume bytes....

Reply:
done

Reply:
done